# Tea plant genomic variation and dispensable genome
Here, we extensively analyzed the genome sequencing data of 363 diverse tea plant accessions, in conjunction with comparative genomics and population genetics methods, to explore the fine population structure and genetic diversity, dispensable genome, SV and PAV patterns in tea plants. We investigated the divergent population or selection patterns of *C.sinensis* var. *sinensis* (CSS) and *C.sinensi*s var. *assamica* (CSA) and identified the PAVs differentially distributed between them. We also verified PAVs might contribute to the divergence of cold tolerance in CSS and CSA. This study not only expanded our understanding of the population structure and diversity of tea plants, but also underscore the importance of SV or PAV on the trait diversification of CSS and CSA, providing a solid genomic foundation for future molecular breeding in cultivated tea plants.

# Content
Pipelines and relevant scripts for:
- Variation and Population structure
- Demographic inference
- Genetic diversity and selection signatures
- Structural variation
- Dispensable genome and PAV
- Phylogenetics

# Citation

# Contacts
Enhua Xia (xiaenhua@gmail.com)

Wei Tong  (wtong@ahau.edu.cn)

